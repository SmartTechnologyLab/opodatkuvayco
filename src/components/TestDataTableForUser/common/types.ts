import type { Deal } from '@/components/common/DataTable/mocks'
import { TableSize } from './constants'

export type TableSizes = TableSize.LG | TableSize.SM

export type RecalculateDeal = (deal: Deal) => void
