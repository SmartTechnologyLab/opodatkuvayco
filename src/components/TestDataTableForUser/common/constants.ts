import { Currency } from '@/constants/currencies'

export const notEditableColumns: string[] = [
  'total',
  'percent',
  'sale.uah',
  'sale.sum',
  'purchase.uah',
  'sale.uah',
  'purchase.sum',
  'sale.rate',
  'purchase.rate'
]

export enum TableSize {
  LG = 'lg',
  SM = 'sm'
}

export const currencyOptions = Object.values(Currency).filter((currency) => currency !== Currency.UAH)

export const tableSizeOptions = Object.values(TableSize)
