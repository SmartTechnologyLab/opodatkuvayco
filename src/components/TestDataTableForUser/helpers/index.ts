import { getCurrencyExchange } from '@/api/getCurrencyExchange'
import type { Deal } from '@/components/common/DataTable/mocks'
import type { Currency } from '@/constants/currencies'
import type { RecalculateDeal } from '@/components/TestDataTableForUser/common/types'
import { clone } from 'ramda'

export const groupAndSumByTicker = (deals: Deal[]) => {
  const copyDeals = clone(deals) as Deal[]

  const grouped = copyDeals.reduce(
    (acc, deal) => {
      if (!acc[deal.ticker]) {
        acc[deal.ticker] = deal
      } else {
        acc[deal.ticker].sale.uah += deal.sale.uah
        acc[deal.ticker].purchase.uah += deal.purchase.uah
        acc[deal.ticker].total += deal.total
      }

      return acc
    },
    {} as Record<string, Deal>
  )

  return Object.values(grouped)
}

export const recalculateVariables = (deal: Deal) => {
  deal.purchase.sum = deal.purchase.price * deal.quantity
  deal.sale.sum = deal.sale.price * deal.quantity
  deal.purchase.uah =
    (deal.purchase.sum + deal.purchase.commission) * deal.purchase.rate + deal.sale.commission * deal.sale.rate
  deal.sale.uah = deal.sale.sum * deal.sale.rate
  deal.total = deal.sale.uah - deal.purchase.uah
  deal.percent = deal.sale.uah / deal.purchase.uah - 1
}

export const updateDealRates = async (deal: Deal, currency: Currency, recalulateDeals: RecalculateDeal) => {
  const [purchase, sale] = await Promise.all([
    getCurrencyExchange(currency, String(deal.purchase.date)),
    getCurrencyExchange(currency, String(deal.sale.date))
  ])

  deal.purchase.rate = purchase.rate
  deal.sale.rate = sale.rate
  recalulateDeals(deal)
}
