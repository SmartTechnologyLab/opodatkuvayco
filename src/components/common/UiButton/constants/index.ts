export enum Icons {
  CHECK = 'pi pi-check',
  USER = 'pi pi-user',
  CROSS = 'pi pi-times',
  DELETE = 'pi pi-trash',
  PLUS = 'pi pi-plus',
  EDIT = 'pi pi-pencil',
  IMPORT = 'pi pi-file-arrow-up'
}

export enum Severities {
  SECONDARY = 'secondary',
  HELP = 'help',
  DANGER = 'danger'
}
