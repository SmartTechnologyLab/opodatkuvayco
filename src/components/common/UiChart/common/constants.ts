const documentStyle = getComputedStyle(document.body)

export const CSS_VARIABLES = [
  documentStyle.getPropertyValue('--blue-200'),
  documentStyle.getPropertyValue('--green-200'),
  documentStyle.getPropertyValue('--yellow-200')
]

export const DEFAULT_CHART_OPTIONS = {
  maintainAspectRatio: false
}
