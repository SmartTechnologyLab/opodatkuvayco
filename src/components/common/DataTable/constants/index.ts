import { _t } from '@/i18n'
import type { TableHeaders } from '../types'
import { Currency } from '@/constants/currencies'

export enum FormatType {
  Date,
  Number,
  CurrencyUAH,
  CurrencyUSD,
  CurrencyEUR,
  Percent
}

export const dynamicCurrencies = [FormatType.CurrencyUSD, FormatType.CurrencyEUR]

export const currenciesName: Record<Currency, Currency> = {
  [Currency.UAH]: Currency.UAH,
  [Currency.USD]: Currency.USD,
  [Currency.EUR]: Currency.EUR
}

export const resultHeaders: TableHeaders[] = [
  {
    header: _t('table.headers.ticker'),
    field: 'ticker'
  },
  {
    header: _t('table.headers.purchase-date'),
    field: 'purchase.date',
    type: FormatType.Date
  },
  {
    header: _t('table.headers.quantity'),
    field: 'quantity',
    type: FormatType.Number
  },
  {
    header: _t('table.headers.purchase'),
    field: 'purchase.price',
    type: FormatType.CurrencyUSD
  },
  {
    header: _t('table.headers.sum'),
    field: 'purchase.sum',
    type: FormatType.CurrencyUSD
  },
  {
    header: _t('table.headers.comission'),
    field: 'purchase.commission',
    type: FormatType.CurrencyUSD
  },
  {
    header: _t('table.headers.rate'),
    field: 'purchase.rate',
    type: FormatType.CurrencyUAH
  },
  {
    header: _t('table.headers.purchase'),
    field: 'purchase.uah',
    type: FormatType.CurrencyUAH
  },
  {
    header: _t('table.headers.sale-date'),
    field: 'sale.date',
    type: FormatType.Date
  },
  {
    header: _t('table.headers.price'),
    field: 'sale.price',
    type: FormatType.CurrencyUSD
  },
  {
    header: _t('table.headers.sum'),
    field: 'sale.sum',
    type: FormatType.CurrencyUSD
  },
  {
    header: _t('table.headers.comission'),
    field: 'sale.commission',
    type: FormatType.CurrencyUSD
  },
  {
    header: _t('table.headers.rate'),
    field: 'sale.rate',
    type: FormatType.CurrencyUAH
  },
  {
    header: _t('table.headers.sale'),
    field: 'sale.uah',
    type: FormatType.CurrencyUAH
  },
  {
    header: _t('table.headers.total'),
    field: 'total',
    type: FormatType.CurrencyUAH
  },
  {
    header: _t('table.headers.percent'),
    field: 'percent',
    type: FormatType.Percent
  }
]
