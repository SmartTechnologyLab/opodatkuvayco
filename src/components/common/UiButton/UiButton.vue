<script setup lang="ts">
import Button from 'primevue/button'
import { computed, useAttrs } from 'vue'
import { Icons, Severities } from './constants'

const attrs = useAttrs()

const props = defineProps<{
  icon?: Icons
  severity?: Severities
  rounded?: boolean
  outlined?: boolean
  raised?: boolean
  class?: string
}>()

const routerLinkTag = 'router-link'
const linkTag = 'a'

const component = computed(() => {
  if (attrs.to) {
    return routerLinkTag
  }
  if (attrs.href) {
    return linkTag
  }

  return null
})
</script>

<template>
  <template v-if="component">
    <Component :is="component">
      <Button v-bind="{ ...props, ...attrs }">
        <slot />
      </Button>
    </Component>
  </template>

  <template v-else>
    <Button v-bind="{ ...props, ...attrs }">
      <slot />
    </Button>
  </template>
</template>
