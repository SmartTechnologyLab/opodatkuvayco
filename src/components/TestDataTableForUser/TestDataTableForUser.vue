<script setup lang="ts">
import DataTable from '@/components/common/DataTable/DataTable.vue'
import { resultHeaders } from '@/components/common/DataTable/constants'
import { getDeal } from '@/components/common/DataTable/mocks'
import type { DataTableCellEditCompleteEvent } from 'primevue/datatable'
import { assocPath } from 'ramda'
import { computed, ref, watch } from 'vue'
import {
  currencyOptions,
  notEditableColumns as notEditableColumnsLargeTable,
  TableSize,
  tableSizeOptions
} from '@/components/TestDataTableForUser/common/constants'
import { useI18n } from 'vue-i18n'
import { getCurrencyExchange } from '@/api/getCurrencyExchange'
import UiButton from '@/components/common/UiButton/UiButton.vue'
import { Currency } from '@/constants/currencies'
import UiSelectButton from '@/components/common/UiSelectButton/UiSelectButton.vue'
import { Icons } from '@/components/common/UiButton/constants'
import type { Currencies } from '@/components/common/DataTable/types'
import type { TableSizes } from '@/components/TestDataTableForUser/common/types'
import { groupAndSumByTicker, recalculateVariables, updateDealRates } from '@/components/TestDataTableForUser/helpers'

const { t } = useI18n()

const rowData = ref({
  ticker: 'DAL',
  quantity: 2,
  purchaseRate: 29.2549,
  purchasePrice: 28.88,
  purchaseCommission: 1.49,
  saleRate: 36.5686,
  salePrice: 42.42,
  saleCommission: 1.62
})

const selectedCurrency = ref<Currencies>(Currency.USD)
const selectedTableSize = ref<TableSizes>(TableSize.LG)
const addBtnText = ref(t('table.btnAddRow'))

const originalData = ref([getDeal(rowData.value), getDeal(), getDeal(), getDeal(), getDeal()])

const headers = computed(() => {
  if (selectedTableSize.value === TableSize.LG) {
    return resultHeaders
  }
  return resultHeaders.filter((header) => header.size?.includes(TableSize.SM))
})

const table = ref({
  headers: headers.value,
  data: [...originalData.value]
})

const highlightedCells = ref(new Set<string>())

const notEditableColumns = computed(() => {
  if (selectedTableSize.value === TableSize.LG) {
    return notEditableColumnsLargeTable
  }
  return headers.value.map((header) => header.field)
})

const highlightCell = (index: number, field: string) => {
  const key = `${index}-${field}`
  highlightedCells.value.add(key)
  setTimeout(() => {
    highlightedCells.value.delete(key)
  }, 2000)
}

const onCellEditComplete = async (event: DataTableCellEditCompleteEvent) => {
  let { newValue, field, index } = event

  if (newValue) {
    if (field === 'purchase.date') {
      const exchangeRate = await getCurrencyExchange(selectedCurrency.value, newValue)
      table.value.data[index].purchase.rate = exchangeRate.rate
    }

    if (field === 'sale.date') {
      const exchangeRate = await getCurrencyExchange(selectedCurrency.value, newValue)
      table.value.data[index].sale.rate = exchangeRate.rate
    }

    const editedRow = table.value.data[index]
    const fieldPath = field.split('.')
    const updatedRow = assocPath(fieldPath, newValue, editedRow)
    table.value.data[index] = updatedRow
    recalculateVariables(updatedRow)

    highlightCell(index, field)
  }
}

const handleAddRow = () => {
  if (originalData.value.length < 8) {
    const newDeal = getDeal()
    originalData.value = [...originalData.value, newDeal]
    table.value.data = [...table.value.data, newDeal]
  }
}

const handleDeleteRow = (rowIndex: number) => {
  originalData.value = originalData.value.filter((_, index) => index !== rowIndex)
  table.value.data = table.value.data.filter((_, index) => index !== rowIndex)
}

watch(
  () => table.value.data.length < 8,
  () => (addBtnText.value = t('table.btnYourLimitLeft'))
)

watch(
  () => selectedCurrency.value,
  async () => {
    await Promise.all(
      table.value.data.map((deal) => updateDealRates(deal, selectedCurrency.value, () => recalculateVariables(deal)))
    )
  }
)

watch(
  () => selectedTableSize.value,
  (newValue) => {
    table.value.headers = headers.value

    if (newValue === TableSize.SM) {
      table.value.data = groupAndSumByTicker(table.value.data)
    } else {
      table.value.data = [...originalData.value]
    }
  }
)

watch(
  () => table.value.data,
  () => {
    if (table.value.data.length && selectedTableSize.value === TableSize.LG) {
      originalData.value = [...table.value.data]
    }
  },
  {
    deep: true
  }
)
</script>

<template>
  <DataTable
    v-if="selectedTableSize === TableSize.LG"
    class="data-table data-table--lg"
    :table="table"
    @onCellEdit="onCellEditComplete($event)"
    :notEditableColumns="notEditableColumns"
    edit-mode="cell"
    :currency="selectedCurrency"
    stripedRows
    removableSort
    rowHover
    scrollable
    key="large-table"
  >
    <template #header>
      <div class="data-table__header">
        <h1 class="data-table__title">{{ t('table.title') }}</h1>
        <UiSelectButton v-model="selectedCurrency" :options="currencyOptions" :allow-empty="false" />
        <UiSelectButton v-model="selectedTableSize" :options="tableSizeOptions" :allow-empty="false" />
      </div>
    </template>

    <template #deleteRow="{ index }">
      <UiButton @click="handleDeleteRow(index)" class="data-table__delete-btn" outlined :icon="Icons.DELETE" />
    </template>

    <template #addRow>
      <UiButton @click="handleAddRow" class="data-table__add-btn" :label="addBtnText" :icon="Icons.PLUS" />
    </template>
  </DataTable>

  <DataTable
    v-else
    :table="table"
    :notEditableColumns="notEditableColumns"
    class="data-table data-table--sm"
    :currency="selectedCurrency"
    stripedRows
    removableSort
    rowHover
    scrollable
    key="small-table"
  >
    <template #header>
      <div class="data-table__header">
        <h1 class="data-table__title">{{ t('table.title') }}</h1>
        <UiSelectButton v-model="selectedTableSize" :options="tableSizeOptions" :allow-empty="false" />
      </div>
    </template>
  </DataTable>
</template>

<style scoped lang="scss">
.data-table {
  min-width: 40vw;
  max-width: 90vw;

  &__header {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  &__title {
    font-size: 2rem;
  }

  &__add-btn {
    margin-left: auto;
    margin-top: 1rem;
    min-width: 180px;
    max-width: 200px;
    display: flex;
    align-items: center;
    justify-content: center;
    gap: 0.5rem;
  }

  &__delete-btn {
    display: flex;
    justify-content: center;
    padding-inline: 10px;
  }
}
</style>
