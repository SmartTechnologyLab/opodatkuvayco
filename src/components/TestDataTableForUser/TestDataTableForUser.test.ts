import { getCurrencyExchange } from '@/api/getCurrencyExchange'
import { stubComponent } from '@/helpers/testHelpers/createComponent'
import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest'
import _DataTable from '@/components/common/DataTable/DataTable.vue'
import { shallowMount, type VueWrapper } from '@vue/test-utils'
import { compose, symmetricDifference, isEmpty } from 'ramda'
import TestDataTableForUser from '@/components/TestDataTableForUser/TestDataTableForUser.vue'
import _UiSelectButton from '@/components/common/UiSelectButton/UiSelectButton.vue'
import type { Currencies, TableHeaders } from '@/components/common/DataTable/types'
import UiButton from '@/components/common/UiButton/UiButton.vue'
import type { TableSizes } from '@/components/TestDataTableForUser/common/types'
import { TableSize } from '@/components/TestDataTableForUser/common/constants'
import { Currency } from '@/constants/currencies'
import type { Deal } from '../common/DataTable/mocks'

vi.mock('@/api/getCurrencyExchange', () => ({
  getCurrencyExchange: vi.fn().mockResolvedValue({ rate: 44.84 })
}))

const DataTable = stubComponent(_DataTable, {
  template: `
    <div>
     <slot name="header" />
     <slot name="deleteRow" :index="0" />
     <slot name="addRow" />
    </div>
  `
})

const UiSelectButton = stubComponent(_UiSelectButton)

const isArraysEqual = compose(isEmpty, symmetricDifference)

describe('Test DataTable for user', () => {
  let wrapper: VueWrapper<InstanceType<typeof TestDataTableForUser>>

  beforeEach(() => {
    wrapper = shallowMount(TestDataTableForUser, {
      global: {
        stubs: {
          DataTable,
          UiSelectButton,
          UiButton
        }
      }
    })
  })

  afterEach(() => {
    wrapper.unmount()
  })

  const findSelectButton = <T>(modelValue: T) =>
    wrapper.findAllComponents(UiSelectButton).find((btn) => btn.attributes().modelvalue === modelValue)

  it('Select currency button has certain params', () => {
    const OPTIONS = `${Currency.USD},${Currency.EUR}`

    const CurrencyButton = findSelectButton<Currencies>(Currency.USD)

    expect(CurrencyButton?.attributes().modelvalue).toStrictEqual(Currency.USD)
    expect(CurrencyButton?.attributes().options).toStrictEqual(OPTIONS)
  })

  it('Changes table size to small with select button', async () => {
    let SelectSizeButton = findSelectButton<TableSizes>(TableSize.LG)

    await SelectSizeButton?.setValue(TableSize.SM)
    await SelectSizeButton?.trigger('click')

    SelectSizeButton = findSelectButton(TableSize.SM)

    expect(SelectSizeButton?.exists()).toBeTruthy()
    expect(SelectSizeButton?.attributes().modelvalue).toStrictEqual(TableSize.SM)
  })

  it('Small table contains particular columns', async () => {
    const SMALL_TABLE_HEADERS = ['ticker', 'sale.uah', 'purchase.uah', 'total']
    let SelectSizeButton = findSelectButton<TableSizes>(TableSize.LG)

    const SmallDataTable = wrapper.findComponent(DataTable)

    await SelectSizeButton?.setValue(TableSize.SM)
    await SelectSizeButton?.trigger('click')

    SelectSizeButton = findSelectButton(TableSize.SM)

    const CURRENT_HEADERS = (SmallDataTable.props().table.headers as TableHeaders[]).map((header) => header.field)

    expect(SelectSizeButton?.attributes().modelvalue).toStrictEqual(TableSize.SM)
    expect(isArraysEqual(CURRENT_HEADERS, SMALL_TABLE_HEADERS)).toBeTruthy()
  })

  it('Changes currency with select button', async () => {
    const SelectCurrencyButton = findSelectButton<Currencies>(Currency.USD)

    await SelectCurrencyButton?.setValue(Currency.EUR)
    await SelectCurrencyButton?.trigger('click')

    expect(SelectCurrencyButton?.attributes().modelvalue).toStrictEqual(Currency.EUR)
  })

  it('Select Buttons must have attribute of allowEmpty to false', () => {
    const SelectCurrencyButton = findSelectButton<Currencies>(Currency.USD)
    const SelectSizeButton = findSelectButton<TableSizes>(TableSize.LG)

    expect(SelectCurrencyButton?.attributes().allowempty).toStrictEqual('false')
    expect(SelectSizeButton?.attributes().allowempty).toStrictEqual('false')
  })

  const ADD_BTN_TEXT = 'Добавити рядок'
  it('Add button attrs', () => {
    const PLUS_ICON = 'pi pi-plus'

    const addButton = wrapper.findAllComponents(UiButton).find((btn) => btn.attributes().label === ADD_BTN_TEXT)

    expect(addButton?.attributes().icon).toEqual(PLUS_ICON)
    expect(addButton?.attributes().label).toEqual(ADD_BTN_TEXT)
  })

  it('Clicking add button adds new row', async () => {
    const addButton = wrapper.findAllComponents(UiButton).find((btn) => btn.attributes().label === ADD_BTN_TEXT)
    const dataTable = wrapper.findComponent(DataTable)

    expect(dataTable.props().table.data.length).toStrictEqual(1)

    await addButton?.trigger('click')

    expect(dataTable.props().table.data.length).toStrictEqual(2)
  })

  it('Clicking delete icon should delete row', async () => {
    const DELETE_ICON = 'pi pi-trash'
    const dataTable = wrapper.findComponent(DataTable)
    const deleteButton = wrapper.findAllComponents(UiButton).find((btn) => btn.attributes().icon === DELETE_ICON)

    await deleteButton?.trigger('click')

    expect(dataTable.props().table.data.length).toStrictEqual(0)
  })

  it('Large data table must have some not editable columns', () => {
    const notEditableCols = [
      'total',
      'percent',
      'sale.uah',
      'sale.sum',
      'purchase.uah',
      'sale.uah',
      'purchase.sum',
      'sale.rate',
      'purchase.rate'
    ]

    const dataTable = wrapper.findComponent(DataTable)

    expect(isArraysEqual(dataTable.props().notEditableColumns, notEditableCols)).toBeTruthy()
  })

  it('Small data table have all columns uneditable', async () => {
    const SMALL_TABLE_HEADERS = ['ticker', 'sale.uah', 'purchase.uah', 'total']
    const selectButton = findSelectButton<TableSizes>(TableSize.LG)

    await selectButton?.setValue(TableSize.SM)
    await selectButton?.trigger('click')

    const dataTable = wrapper.findComponent(DataTable)

    expect(isArraysEqual(dataTable.props().notEditableColumns, SMALL_TABLE_HEADERS))
  })

  it('Fetched and changed rates correctly', async () => {
    const MOCKED_RATE = 44.84
    const dates: string[] = []

    const selectButton = findSelectButton<Currencies>(Currency.USD)
    const dataTable = wrapper.findComponent(DataTable)

    const deals: Deal[] = dataTable.props().table.data

    deals.forEach((deal) => {
      dates.push(String(deal.purchase.date), String(deal.sale.date))
    })

    await selectButton?.setValue(Currency.EUR)
    await selectButton?.trigger('click')

    dates.forEach((date) => {
      expect(getCurrencyExchange).toHaveBeenCalledWith(Currency.EUR, date)
    })

    deals.forEach((deal) => {
      expect(deal.purchase.rate).toStrictEqual(MOCKED_RATE)
      expect(deal.sale.rate).toStrictEqual(MOCKED_RATE)
    })
  })
})
