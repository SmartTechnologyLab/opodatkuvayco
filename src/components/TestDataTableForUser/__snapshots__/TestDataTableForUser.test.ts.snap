// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test DataTable for user > Add button attrs 1`] = `"pi pi-plus"`;

exports[`Test DataTable for user > Add button attrs 2`] = `"Добавити рядок"`;

exports[`Test DataTable for user > Changes currency with select button 1`] = `"EUR"`;

exports[`Test DataTable for user > Changes table size to small with select button 1`] = `true`;

exports[`Test DataTable for user > Changes table size to small with select button 2`] = `"Short Statement"`;

exports[`Test DataTable for user > Fetched and changed rates correctly 1`] = `"EUR"`;

exports[`Test DataTable for user > Fetched and changed rates correctly 2`] = `44.84`;

exports[`Test DataTable for user > Fetched and changed rates correctly 3`] = `44.84`;

exports[`Test DataTable for user > Large data table must have some not editable columns 1`] = `true`;

exports[`Test DataTable for user > Select Buttons must have attribute of allowEmpty to false 1`] = `"false"`;

exports[`Test DataTable for user > Select Buttons must have attribute of allowEmpty to false 2`] = `"false"`;

exports[`Test DataTable for user > Select currency button has certain params 1`] = `"USD"`;

exports[`Test DataTable for user > Select currency button has certain params 2`] = `"USD,EUR"`;

exports[`Test DataTable for user > Small data table group tickers by ticker name and sum it data 1`] = `1`;

exports[`Test DataTable for user > Small data table group tickers by ticker name and sum it data 2`] = `1`;

exports[`Test DataTable for user > Small data table group tickers by ticker name and sum it data 3`] = `398`;

exports[`Test DataTable for user > Small data table have all columns uneditable 1`] = `true`;

exports[`Test DataTable for user > Small table contains particular columns 1`] = `"Short Statement"`;

exports[`Test DataTable for user > Small table contains particular columns 2`] = `true`;
